### ISSUE TITLE
tests/db_instance: Panic at the Terraform v0.15.0-alpha

### ISSUE THREAD

**Issue by YakDriver** [hashicorp/terraform-provider-aws#17770]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Acceptance Test

`TestAccAWSDBInstance_IsAlreadyBeingDeleted ` panics on Terraform v0.15.0-alpha:

```
    resource_aws_db_instance_test.go:483: Step 2/2 error: Error running destroy: exit status 2
        panic: runtime error: invalid memory address or nil pointer dereference
        [signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0xb173ba]
        
        goroutine 11 [running]:
        github.com/hashicorp/terraform/configs/configload.(*Loader).ImportSourcesFromSnapshot(0xc00009d5e0, 0x0)
        	/home/circleci/project/project/configs/configload/loader.go:143 +0x3a
        github.com/hashicorp/terraform/backend/local.(*Local).context(0xc0005cbc70, 0xc0001a8b60, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0)
        	/home/circleci/project/project/backend/local/backend_local.go:108 +0x8d2
        github.com/hashicorp/terraform/backend/local.(*Local).opApply(0xc0005cbc70, 0x2ca1aa0, 0xc0002529c0, 0x2ca1aa0, 0xc000252a00, 0xc0001a8b60, 0xc000252980)
        	/home/circleci/project/project/backend/local/backend_apply.go:51 +0x2bf
        github.com/hashicorp/terraform/backend/local.(*Local).Operation.func1(0xc00037da50, 0xc00037da60, 0xc00037da70, 0xc0005cbc70, 0xc00037da40, 0x2ca1aa0, 0xc0002529c0, 0x2ca1aa0, 0xc000252a00, 0xc0001a8b60, ...)
        	/home/circleci/project/project/backend/local/backend.go:345 +0xe8
        created by github.com/hashicorp/terraform/backend/local.(*Local).Operation
        	/home/circleci/project/project/backend/local/backend.go:339 +0x36d
--- FAIL: TestAccAWSDBInstance_IsAlreadyBeingDeleted (410.17s)
```

### Affected Resources

* aws_db_instance

### Terraform Configuration Files

Please include all Terraform configurations required to reproduce the bug. Bug reports without a functional reproduction may be closed without investigation.

```hcl
resource "aws_db_instance" "test" {
  allocated_storage   = 5
  engine              = data.aws_rds_orderable_db_instance.test.engine
  identifier          = "yakstance"
  instance_class      = data.aws_rds_orderable_db_instance.test.instance_class
  password            = "avoid-plaintext-passwords"
  username            = "tfacctest"
  skip_final_snapshot = true
}
```

### References

* https://github.com/hashicorp/terraform/releases/tag/v0.15.0-alpha20210210

**Comment by apparentlymart** [comment]
Hi @YakDriver!

Do you know if the test harness that ran this test works by creating a saved plan (using `terraform plan -out=...` and then applying the saved file) or if it just runs `terraform apply` directly to plan and apply in a single shot?

It seems like something's gone wrong with the plan loading here and caused the configuration snapshot to not be populated, so I'm trying to figure out how best to replicate what actions the test harness would've taken here. I would not expect this error to depend on any particular configuration _contents_ because it seems to be failing in code that would run regardless of what is present in the configuration files.

**Comment by apparentlymart** [comment]
(Regardless of the details I expect this _is_ a Terraform CLI regression, so if you'd like to escalate it into an issue in the hashicorp/terraform repository then we could continue digging into it over there instead.)

**Comment by jbardin** [comment]
Looking at the state trace, this is loading the config snapshot from a plan file.
We don't see what the error is here because of the panic, but this codepath definitely misses an error check and can pass a nil snapshot to the configloader

**Comment by jbardin** [comment]
As a follow up, these configload panics appear to have been fixed in the latest beta (at least the ones I tried no longer panic). I'm not sure what the root cause was, but if it still can happen, the reason for the panic itself is fixed in the 0.15 branch and we'll see what the underlying error is.

**Comment by YakDriver** [comment]
Thanks for the insights! I will compare with the beta results to make sure that all is as expected.
